File: Map/Z_InterpretorWPro/src/statement/CompStatement.java
package statement;

import domain.IStack;
import domain.PrgState;

public class CompStatement implements IStatement
{
	private IStatement s1,s2;
	
	public CompStatement(IStatement st1, IStatement st2)
	{
		s1=st1;
		s2=st2;
	}

	@Override
	public PrgState execute(PrgState ps) 
	{
		IStack<IStatement> stack = ps.getExeStack();
		stack.push(s2);
		stack.push(s1);
		return null;
	}
	
	@Override
	public String toString()
	{
		return "("+s1+" ; "+s2+")";
	}
	
	

}

File: Map/Map/Map/statement/CompStatement.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Map.domain;

namespace Map.statement
{
    class CompStatement : IStatement
    {
        private IStatement st1, st2;

        public CompStatement(IStatement s1,IStatement s2)
        {
            st1 = s1;
            st2 = s2;
        }

        public ProgramState execute(ProgramState ps)
        {
            ps.exeStack.push(st2);
            ps.exeStack.push(st1);
            return null;
        }

        public override string ToString()
        {
            return "(" + st1 + ";" + st2 + ")";
        }
    }
}
